workspaces/marketplace/plugins/marketplace-common/src/utils.ts

```diff
@@ -0,0 +1,136 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  EntityFilterQuery,
+  EntityOrderQuery,
+  QueryEntitiesRequest,
+} from '@backstage/catalog-client/index';
+import { GetPluginsRequest, SortOrder } from './types';
+
+const DefaultOrderFields: EntityOrderQuery = [
+  { field: 'metadata.name', order: 'asc' },
+];
+const DefaultLimit = 20;
+const RequiredFilter = { kind: 'plugin' };
+
+export const convertGetPluginsRequestToQueryEntitiesRequest = (
+  query?: GetPluginsRequest,
+): QueryEntitiesRequest => {
+  let orderFields: EntityOrderQuery = DefaultOrderFields;
+
+  if (query?.orderFields) {
+    if (Array.isArray(query.orderFields)) {
+      orderFields = query.orderFields.map(field => {
+        if (typeof field === 'string') {
+          const [key, order] = (field as string).split(',');
+          return { field: key, order: order as SortOrder };
+        }
+        return field;
+      });
+    } else {
+      if (typeof query.orderFields === 'string') {
+        const [key, order] = (query.orderFields as string).split(',');
+        orderFields = { field: key, order: order as SortOrder };
+      }
+    }
+  }
+
+  const filter: EntityFilterQuery = query?.filter
+    ? {
+        ...(typeof query.filter === 'string'
+          ? JSON.parse(query.filter)
+          : query.filter),
+        ...RequiredFilter,
+      }
+    : RequiredFilter;
+  const payload: QueryEntitiesRequest = {
+    filter: filter,
+    orderFields: orderFields,
+    limit: query?.limit ? Number(query.limit) : DefaultLimit,
+    offset: query?.offset ? Number(query.offset) : undefined,
+  };
+  if (query?.searchTerm) {
+    payload.fullTextFilter = { term: query.searchTerm };
+  }
+  return payload;
+};
+
+export const convertGetPluginRequestToSearchParams = (
+  query?: GetPluginsRequest,
+): URLSearchParams => {
+  const params = new URLSearchParams();
+  if (query?.limit) params.append('limit', String(query.limit));
+  if (query?.offset) params.append('offset', String(query.offset));
+  if (query?.searchTerm) params.append('searchTerm', query.searchTerm);
+  if (query?.orderFields) {
+    if (Array.isArray(query.orderFields)) {
+      query.orderFields.forEach(({ field, order }) => {
+        params.append('orderFields', `${field},${order}`);
+      });
+    } else {
+      const { field, order } = query.orderFields;
+      params.append('orderFields', `${field},${order}`);
+    }
+  }
+  if (query?.filter) params.append('filter', JSON.stringify(query.filter));
+  return params;
+};
+
+export const convertSearchParamsToGetPluginsRequest = (
```

Comment:
I think it will be good to extend the generic `encodeQueryParams` function to handle more fields, Instead of having it's own version for GetPluginsRequest. This way we can reuse the function for other API like PluginLists, Packages etc.